subiquity/server/server.py

```diff
@@ -396,7 +404,8 @@ def make_apport_report(self, kind, thing, *, wait=False, **kw):
         return self.error_reporter.make_apport_report(kind, thing, wait=wait, **kw)
 
     async def _run_error_cmds(self, report):
-        await report._info_task
+        if hasattr(report, "_info_task"):
```

Comment:
I'd like you to double-check this line.
`ErrorReport` unconditionally creates the _info_task attribute, so it should either be None or an asyncio task.   So it's always there, but may be None, and `await None` is bad.  Should this be `if report._info_task is not None` instead?

Reply:
I didn't see the default value for the attribute, so your recommendation is right if `report` is of type `ErrorReport`. Although in the case of `AutoinstallError` exceptions I'm passing `report=None`. I was trying to catch both cases here, but I should have been more explicit.

My assumption is that the `ErrorReport` class is about generating apport reports based on the exception, but here we don't want to do that.

What do you think of rewriting as:
```python
async def _run_error_cmds(self, report: Optional[ErrorReport] = None):
    if report is not None and report._info_task is not None:
        await report._info_task
     ...
```

Reply:
Ah, report=None would also have been hosed (`getattr(None, ...)` is also bad.)
I expect that ErrorReport is the correct type here, so your latest suggestion looks correct.